config_parser.cc

```diff
@@ -152,6 +152,7 @@ bool NginxConfigParser::Parse(std::istream* config_file, NginxConfig* config) {
   config_stack.push(config);
   TokenType last_token_type = TOKEN_TYPE_START;
   TokenType token_type;
+  int counter = 0;
```

Comment:
I agree with McKenna, calling it something like "balancedBlocks" would be a lot more useful, and maybe leaving a comment nearby.